cardano-explorer/src/Explorer/Web/Server.hs

```diff
@@ -179,90 +192,116 @@ getBlocksPagesTotal backend mPageSize = do
     (_, True) -> pure $ Left $ Internal "Page size must be greater than 1 if you want to display blocks."
     _ -> pure $ Right $ roundToBlockPage blocksTotal
 
+hexToBytestring :: T.Text -> ExceptT ExplorerError Handler BS.ByteString
+hexToBytestring text = do
+  case B16.decode (SB8.pack (T.unpack text)) of
+    (blob, "") -> pure blob
+    _ -> throwE $ Internal "cant parse hex"
+
+bytestringToText :: BS.ByteString -> T.Text
+bytestringToText = T.pack . SB8.unpack . B16.encode
```

Comment:
`T.pack . SB8.unpack` can be replaced with `T.decodeUtf8` (in `Data.Text.Encoding`).